### PR TITLE
krb5-tests.py: add distribution specific test cases

### PR DESCRIPTION
**Pull request by nasastry**
Add distribution specific test cases.
With the support of yaml, can run distribution and upstream test cases.

Signed-off-by: Nageswara R Sastry <rnsastry@linux.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
```
RHEL:
JOB ID     : 8c1a10ede10ead11ad2f9d5e2259d4dffccea502
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-21T00.21-8c1a10e/job.log
 (1/2) krb5-tests.py:Krb5.test;run-run_type-upstream-ca87:  FAIL: 'make check' failed. (51.98 s)
 (2/2) krb5-tests.py:Krb5.test;run-run_type-distro-a8a9:  ERROR: krb5 source install failed. (4.88 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-21T00.21-8c1a10e/results.html
JOB TIME   : 64.93 s

SLES:
JOB ID     : fd14a9ae050525514ecc990c6e4f4100607a5280
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-09-21T10.57-fd14a9a/job.log
 (1/2) krb5-tests.py:Krb5.test;run-run_type-upstream-ca87: FAIL: 'make check' failed. (42.46 s)
 (2/2) krb5-tests.py:Krb5.test;run-run_type-distro-a8a9: FAIL: 'make check' failed. (51.68 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-09-21T10.57-fd14a9a/results.html
JOB TIME   : 115.18 s
```

File: security/krb5-tests.py [hunk 1]
```diff
@@ -35,22 +35,35 @@ def setUp(self):
...
+            if not self.srcdir:
+                self.fail("krb5 source install failed.")
 
     def test(self):
         '''
         Running tests from krb5
         '''
         count = 0
-        output = build.run_make(self.sourcedir, extra_args="check",
+        output = build.run_make(self.srcdir, extra_args="check",
```
Comment (Naresh-ibm) [inline security/krb5-tests.py:63]: @nasastry are we not doing this twice for upstream type?
as you have added the build.make() in setup for upstream type already.

Comment (nasastry) [reply security/krb5-tests.py:63]: @Naresh-ibm 
In `setup` - `configure and make`
in `test` - running `make check` << this will run the test cases.

**Review by Naresh-ibm — APPROVED** [review]

**Comment by Naresh-ibm** [comment]
LGTM

File: security/krb5-tests.py (hunk 1 above)
Comment (Naresh-ibm) [reply security/krb5-tests.py:63]: ok

**Comment by nasastry** [comment]
```
SLES:
JOB ID     : 6232a769bad7f03e343b2f18b7374fdcbcfdc2d6
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-07T13.37-6232a76/job.log
 (1/2) krb5-tests.py:Krb5.test;run-run_type-upstream-ca87:  FAIL: krb5-tests.py: 'make check' failed. (44.66 s)
 (2/2) krb5-tests.py:Krb5.test;run-run_type-distro-a8a9:  FAIL: krb5-tests.py: 'make check' failed. (54.16 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-07T13.37-6232a76/results.html
JOB TIME   : 120.97 s



RHEL:
JOB ID     : 1f516b15a7ffcf03a7766b5b62a068768e413f6d
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-07T03.10-1f516b1/job.log
 (1/2) krb5-tests.py:Krb5.test;run-run_type-upstream-ca87: FAIL: krb5-tests.py: 'make check' failed. (52.64 s)
 (2/2) krb5-tests.py:Krb5.test;run-run_type-distro-a8a9: ERROR: krb5-tests.py: krb5 source install failed. (5.46 s)
RESULTS    : PASS 0 | ERROR 1 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-07T03.10-1f516b1/results.html
JOB TIME   : 68.26 s
```

File: security/krb5-tests.py [hunk 2]
```diff
@@ -35,22 +35,39 @@ def setUp(self):
...
                 self.cancel('%s is needed for the test to be run' % package)
-        url = "https://github.com/krb5/krb5/archive/master.zip"
-        tarball = self.fetch_asset(url, expire='7d')
-        archive.extract(tarball, self.workdir)
-        self.sourcedir = os.path.join(self.workdir, 'krb5-master/src')
-        os.chdir(self.sourcedir)
-        process.run('autoreconf', ignore_status=True)
-        process.run('./configure', ignore_status=True)
-        build.make(self.sourcedir)
+        run_type = self.params.get('type', default='upstream')
```
Comment (Naresh-ibm) [inline security/krb5-tests.py:38]: @nasastry make distro as default parameter

File: security/krb5-tests.py [hunk 3]
```diff
@@ -35,22 +35,39 @@ def setUp(self):
...
+        if run_type == "upstream":
+            default_url = "https://github.com/krb5/krb5/archive/master.zip"
```
Comment (Naresh-ibm) [inline security/krb5-tests.py:40]: remove the default_url as we are hard_coding it in yaml

File: security/krb5-tests.py [hunk 4]
```diff
@@ -35,22 +35,39 @@ def setUp(self):
...
+            url = self.params.get('url', default=default_url)
```
Comment (Naresh-ibm) [inline security/krb5-tests.py:41]: make default as None here

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

File: security/krb5-tests.py (hunk 4 above)
Comment (abdhaleegit) [reply security/krb5-tests.py:41]: let default point to url, so test wont fail

**Review by abdhaleegit — APPROVED** [review]
LGTM

**Review by Naresh-ibm — COMMENTED** [review]
Thanks @nasastry  and @abdhaleegit

**Review by Naresh-ibm — APPROVED** [review]

**Comment by Naresh-ibm** [comment]
LGTM

**Pull request merged**